### PR TITLE
feat: curioweb: Sector info page

### PR DESCRIPTION
**Pull request by magik6k**
## Proposed Changes
This PR adds an individual sector info page

## Additional Info
Screenshot:
2024-04-06-171140_690x946_scrot

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: lib/harmony/harmonytask/task_type_handler.go
Comment (magik6k) [inline lib/harmony/harmonytask/task_type_handler.go:1]: something went wrong with a rebase here..

File: curiosrc/web/hapi/simpleinfo.go [hunk 1]
```diff
@@ -128,6 +135,252 @@ func (a *app) nodeInfo(writer http.ResponseWriter, request *http.Request) {
...
+	afterSeed := task.SeedEpoch != nil && *task.SeedEpoch <= int64(epoch)
+
+	var sectorLocations []struct {
+		CanSeal, CanStore bool
+		FileType          storiface.SectorFileType `db:"sector_filetype"`
+		StorageID         string                   `db:"storage_id"`
+		Urls              string                   `db:"urls"`
+	}
+
+	err = a.db.Select(ctx, &sectorLocations, `SELECT p.can_seal, p.can_store, l.sector_filetype, l.storage_id, p.urls FROM sector_location l
```
Comment (snadrus) [inline curiosrc/web/hapi/simpleinfo.go:224]: please check err

File: curiosrc/web/hapi/simpleinfo.go [hunk 2]
```diff
@@ -128,6 +135,252 @@ func (a *app) nodeInfo(writer http.ResponseWriter, request *http.Request) {
...
+		FileTypeRowSpan int
+
+		Locations []fileLocations
+	}
+	locs := []locationTable{}
+
+	for i, loc := range sectorLocations {
+		loc := loc
+
+		urlList := strings.Split(loc.Urls, ",") // Assuming URLs are comma-separated
```
Comment (snadrus) [inline curiosrc/web/hapi/simpleinfo.go:247]: Adding SQL comments (if verified) may be useful

File: curiosrc/web/hapi/simpleinfo.go [hunk 3]
```diff
@@ -128,6 +135,252 @@ func (a *app) nodeInfo(writer http.ResponseWriter, request *http.Request) {
...
+		appendNonNil(task.TaskPoRep)
+		appendNonNil(task.TaskFinalize)
+		appendNonNil(task.TaskMoveStorage)
+		appendNonNil(task.TaskCommitMsg)
+
+		if len(taskIDs) > 0 {
+			ids := make([]int64, 0, len(taskIDs))
+			for id := range taskIDs {
+				ids = append(ids, id)
+			}
```
Comment (snadrus) [inline curiosrc/web/hapi/simpleinfo.go:331]: ids := lo.Keys(taskIDs)

**Review by snadrus — CHANGES_REQUESTED** [review]

**Review by snadrus — APPROVED** [review]

**Pull request merged**